File: src/ERP/acme.atena.domain/DTO/Account/Pagamento.cs
using acme.atena.domain.DTO.Enum;
using acme.atena.domain.DTO.Person;
using acme.atena.domain.DTO.Seguranca;
using acme.atena.domain.DTO.Util;
using System;
using System.Collections.Generic;
using System.Text;

namespace acme.atena.domain.DTO.Account
{
    public class Pagamento : AbstractEntityCompetencia
    {
        private int _numeroDaParcela = 0;
        public Pagamento()
        {
            if (!Parcelado)
                _numeroDaParcela = 1;

            if (DataCredito is null || DataCredito == new DateTime(1, 1, 1))
                DataCredito = DataPagamento.AddDays(2);
        }

        public bool Parcelado { get; set; }
        public int NumeroDaParcela { get => _numeroDaParcela; set => _numeroDaParcela = Parcelado ? value : _numeroDaParcela; }
        public DateTime DataPagamento { get; set; }
        public DateTime? DataCredito { get; set; }
        public decimal ValorPago { get; set; }
        public EnumTipoPagamento EnumTipoPagamento { get; set; }
        public Guid? UsuarioId { get; set; }
        public Guid? FornecedorId { get; set; }
        public Guid? ClienteId { get; set; }
        public Guid? EmpresaId { get; set; }
        public Guid? PagamentoVendaId { get; set; }

        public virtual Cliente Cliente { get; set; }
        public virtual Fornecedor Fornecedor { get; set; }
        public virtual Usuario Usuario { get; set; }
        public virtual Empresa Empresa { get; set; }
        public virtual PagamentoVenda PagamentoVenda { get; set; }

        public virtual ICollection<PagamentoFormaPagamento> Pagamentos { get; set; } = new HashSet<PagamentoFormaPagamento>();

    }
}

File: site/coreui-free-angular-admin-template-master/src/app/view-model/response/account/pagamento.ts
import { PagamentoVenda } from './pagamento-venda';
import { UsuarioResponse } from '../security/usuario';
import { AbstractEntityCompetencia } from "../abstract-entity-competencia";
import { EnumTipoPagamento } from "../enum/enum-tipo-pagamento";
import { Cliente } from "../person/cliente";
import { Fornecedor } from "../person/fornecedor";
import { Empresa } from '../person/empresa';
import { PagamentoFormaPagamento } from './pagamento-forma-pagamento';

export class Pagamento extends AbstractEntityCompetencia
{
  public parcelado : boolean;
  public numeroDaParcela : number;
  public dataPagamento: Date;
  public dataCredito: Date;
  public valorPago: number;
  public enumTipoPagamento: EnumTipoPagamento;
  public usuarioId: string;
  public fornecedorId : string;
  public clienteId: string;
  public empresaId: string;
  public pagamentoVendaId: string;

  public  Cliente: Cliente;
  public  Fornecedor: Fornecedor;
  public  Usuario : UsuarioResponse;
  public  Empresa : Empresa;
  public  PagamentoVenda: PagamentoVenda;

  public PagamentosFormasPagamentos:PagamentoFormaPagamento[];

}
